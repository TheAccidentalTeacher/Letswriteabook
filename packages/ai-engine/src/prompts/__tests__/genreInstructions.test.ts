import { genreInstructions } from '../genreInstructions';

describe('genreInstructions', () => {
  it('matches the captured legacy structure', () => {
    expect(genreInstructions).toMatchSnapshot();
  });
});
