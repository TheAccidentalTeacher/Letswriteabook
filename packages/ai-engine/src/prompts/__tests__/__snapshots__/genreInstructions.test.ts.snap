// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genreInstructions matches the captured legacy structure 1`] = `
{
  "ADVENTURE": {
    "EXPLORATION": "Focus on journeys into unknown territories or environments. Create detailed, immersive descriptions of unfamiliar settings. Develop challenges specific to the environment being explored. Consider both physical and psychological impacts of exploration. Balance external challenges with character development.",
    "WILDERNESS_ADVENTURE": "Set in remote natural environments with minimal human presence. Create detailed, immersive descriptions of specific natural settings. Develop challenges arising from wilderness conditions and wildlife. Consider the psychological impact of isolation in nature. Balance environmental challenges with character growth.",
  },
  "CHILDRENS": {
    "CHILDRENS_HUMOR": "Create stories with strong comedic elements appealing to young readers. Develop humor through character, situation, and wordplay appropriate to target age. Focus on universal experiences that children find funny. Consider age-appropriate humor without relying on sarcasm or cynicism. Balance humor with genuine emotional moments.",
    "PICTURE_BOOK": "Create simple, clear stories appropriate for ages 3-8 with illustrations on every page. Develop concise text with strong read-aloud quality. Focus on single, straightforward concepts or messages. Consider how text and illustrations will work together. Balance simplicity with engaging narrative and language play.",
  },
  "CHRISTIAN_FICTION": {
    "AMISH_FICTION": "Focus on Amish communities and their distinct lifestyle and faith practices. Research Amish culture, beliefs, and daily life for authentic portrayal. Develop characters navigating tensions between Amish community and outside world. Consider both challenges and benefits of Amish lifestyle. Balance cultural specificity with universal human experiences.",
    "BIBLICAL_FICTION": "Elaborate on stories from the Bible with fictional details while remaining faithful to scripture. Research historical context of biblical events for authentic setting details. Develop biblical figures as fully-realized characters while respecting scriptural portrayal. Consider theological implications of fictional additions. Balance creative storytelling with biblical accuracy.",
    "CHRISTIAN_FAMILY_SAGA": "Trace family stories across generations with Christian faith as a connecting element. Create family members with varying relationships to faith tradition. Develop how faith is transmitted, questioned, and renewed across generations. Consider how family dynamics interact with religious practice. Balance family drama with spiritual themes and growth.",
    "CHRISTIAN_FANTASY": "Create fantasy worlds that incorporate Christian themes, allegory, or moral framework. Develop fantasy elements that serve to illuminate spiritual truths. Consider how magic or supernatural elements align with Christian worldview. Create clear moral framework while avoiding simplistic good/evil dichotomies. Balance fantasy elements with spiritual meaning.",
    "CHRISTIAN_SCIENCE_FICTION": "Develop science fiction scenarios that explore faith questions in futuristic or technological contexts. Create future worlds where Christianity continues or is challenged in new ways. Consider how faith might respond to scientific advancement or extraterrestrial contact. Develop ethical questions arising from technology through Christian lens. Balance scientific elements with spiritual themes.",
    "CHRISTIAN_WOMENS_FICTION": "Center on women's experiences with faith as a significant element in their lives. Create female protagonists whose Christian beliefs influence their approach to challenges. Develop realistic portrayal of women's roles in various Christian traditions. Consider both traditional and progressive Christian women's perspectives. Balance women's issues with spiritual growth.",
    "CONTEMPORARY_CHRISTIAN_FICTION": "Set in modern times with characters whose Christian faith influences their lives and decisions. Create authentic faith journeys without simplistic solutions to life's problems. Develop characters who struggle with real issues while maintaining or developing faith. Consider how Christianity intersects with contemporary challenges. Balance spiritual content with engaging narrative.",
  },
  "FANTASY": {
    "ARTHURIAN": "Draw inspiration from Arthurian legends and medieval romance. Include knights, quests for honor, courtly love, and moral codes. Focus on chivalry, loyalty, and the tension between duty and desire. Incorporate elements of Celtic mythology.",
    "DARK_FANTASY": "Combine fantasy elements with horror themes. Include darker magic, morally ambiguous characters, and bleak atmospheres. Explore themes of corruption, sacrifice, and the cost of power. Balance fantasy wonder with psychological darkness.",
    "EPIC_FANTASY": "Create vast, complex narratives spanning multiple books or long timeframes. Include detailed world-building, complex magic systems, and large casts of characters. Focus on the fate of kingdoms or worlds hanging in the balance.",
    "HIGH_FANTASY": "Create completely imaginary worlds with their own rules, cultures, and histories. Include magic systems, mythical creatures, and epic quests. Focus on the struggle between good and evil on a grand scale. Develop rich world-building with detailed cultures and languages.",
    "MAGICAL_REALISM": "Incorporate magical elements into realistic settings as natural parts of the world. Characters accept magical events without question. Focus on emotional and psychological depth rather than explaining the magic. Blend the extraordinary with the mundane.",
    "ROMANTIC_FANTASY": "Center romantic relationships within fantasy settings. Balance love stories with magical adventures. Include fantastical obstacles to romance such as curses, different species, or magical duties. Focus on emotional development alongside magical elements.",
    "SWORD_AND_SORCERY": "Focus on individual heroes in magical worlds, often featuring warriors and wizards. Emphasize action and adventure over complex world-building. Include ancient evils, powerful artifacts, and personal quests for revenge or glory.",
    "URBAN_FANTASY": "Set in contemporary world with hidden magical elements. Blend mundane reality with supernatural creatures and magic. Protagonists often discover hidden magical societies. Balance familiar settings with fantastical elements.",
  },
  "HISTORICAL_FICTION": {
    "AMERICAN_CIVIL_WAR": "Set during the American Civil War period (1860s). Include perspectives from different sides and social positions. Research specific battles, political issues, and social changes of the era.",
    "ANCIENT_WORLD": "Set in ancient civilizations such as Rome, Greece, Egypt, or Mesopotamia. Research historical details extensively for authenticity. Include accurate social structures, daily life, and belief systems of the period.",
    "COLONIAL_AMERICA": "Set in American colonial period before independence. Include interactions with Native Americans, religious communities, and challenges of establishing new settlements. Research daily life and survival challenges.",
    "MEDIEVAL": "Set in medieval Europe or other medieval societies. Include feudalism, religious influence, and period-appropriate technology. Research social hierarchies and cultural practices of the specific time and place.",
    "RENAISSANCE": "Set during the Renaissance period with focus on cultural and intellectual flowering. Include art, science, and social change of the period. Research specific Renaissance cities and their unique characteristics.",
    "VICTORIAN": "Set in 19th century Victorian era with attention to social conventions and rapid social change. Include industrial revolution impacts, class distinctions, and evolving social norms.",
    "WORLD_WAR_I": "Set during or around WWI with focus on how the war affected individuals and society. Research specific battles, home front experiences, and post-war consequences. Include authentic details of military and civilian life.",
    "WORLD_WAR_II": "Set during WWII with focus on various perspectives and experiences of the war. Include resistance movements, Holocaust experiences, home front challenges, and liberation. Research specific locations and events.",
  },
  "HORROR": {
    "BODY_HORROR": "Focus on physical transformation, disease, or bodily violation as sources of horror. Include themes of loss of bodily autonomy and fear of physical change or contamination.",
    "COSMIC_HORROR": "Include incomprehensible entities or forces beyond human understanding. Focus on humanity's insignificance in the universe. Create fear through the unknown and unknowable rather than direct threats.",
    "FOLK_HORROR": "Include rural or isolated communities with dark secrets or ancient traditions. Focus on the clash between modernity and primitive beliefs. Include themes of sacrifice and ancient powers.",
    "GOTHIC_HORROR": "Set in atmospheric locations like old mansions, castles, or isolated settings. Include elements of decay, family secrets, and historical darkness. Focus on mood and atmosphere as much as frightening events.",
    "PSYCHOLOGICAL_HORROR": "Create fear through mental and emotional terror rather than external threats. Include unreliable perceptions, madness, and breakdown of reality. Focus on what's happening in characters' minds.",
    "SLASHER": "Feature a killer stalking and systematically eliminating victims. Include survival elements and final girl/final boy archetypes. Focus on tension and suspense leading up to confrontations.",
    "SUPERNATURAL_HORROR": "Include ghosts, demons, or otherworldly entities as sources of fear. Focus on the unknown and unexplainable. Build atmosphere through suggestion and gradual revelation rather than immediate shock.",
    "ZOMBIE_APOCALYPSE": "Feature undead creatures and societal collapse. Focus on survival, human nature under extreme stress, and the breakdown of civilization. Explore themes of what makes us human.",
  },
  "HUMOR": {
    "SATIRE": "Use wit and irony to critique social conventions, institutions, or human folly. Create exaggerated situations that reveal underlying truths about targets of satire. Develop characters that embody particular social types or attitudes. Consider the constructive purpose behind the critique. Balance critical elements with entertainment value.",
    "SLAPSTICK": "Focus on physical comedy, pratfalls, and exaggerated bodily humor. Create elaborate physical gags with careful timing and escalation. Develop characters whose physical traits or skills contribute to comic situations. Consider varying the types of physical humor to maintain freshness. Balance physical comedy with character and plot development.",
  },
  "LITERARY_FICTION": {
    "CHARACTER_STUDY": "Focus intensively on character development and psychology over plot. Explore internal conflicts, personal growth, and human nature. Develop complex, flawed, realistic characters with deep inner lives.",
    "COMING_OF_AGE": "Focus on protagonists transitioning from youth to adulthood. Include loss of innocence, self-discovery, and major life decisions. Explore universal themes of growth and identity formation.",
    "EXPERIMENTAL": "Challenge conventional narrative structures, styles, or formats. Include unusual perspectives, non-linear storytelling, or innovative literary techniques. Push boundaries while maintaining reader engagement.",
    "FAMILY_SAGA": "Trace family relationships across multiple generations or extended time periods. Include changing social contexts and their impact on family dynamics. Explore themes of inheritance, tradition, and change.",
    "IMMIGRANT_EXPERIENCE": "Explore themes of cultural displacement, adaptation, and identity in new countries. Include authentic details of specific immigrant experiences. Balance universal themes with culturally specific elements.",
    "LITERARY_REALISM": "Present realistic portrayals of contemporary life without genre elements. Focus on authentic human experiences and social issues. Include detailed observation of daily life and human behavior.",
    "SOCIAL_COMMENTARY": "Explore current social, political, or cultural issues through narrative. Include realistic portrayals of social problems and their human impact. Balance message with storytelling.",
    "WORKPLACE_DRAMA": "Set primarily in professional environments exploring career challenges, office politics, and work-life balance. Include realistic workplace dynamics and professional relationships.",
  },
  "MYSTERY": {
    "COZY_MYSTERY": "Set in small communities with amateur sleuths, minimal violence, and puzzle-like mysteries. Feature recurring characters and local settings. Include elements of daily life and community relationships. Keep violence off-page and focus on clever deduction.",
    "DETECTIVE": "Feature a professional or amateur detective solving crimes through logical deduction. Include red herrings, suspects with motives, and clues that can be followed by readers. Focus on the investigative process and character development of the detective.",
    "HARD_BOILED": "Feature cynical, world-weary detectives in urban settings. Include gritty realism, moral ambiguity, and complex plots involving corruption. Focus on atmosphere and character psychology as much as the mystery itself.",
    "HISTORICAL_MYSTERY": "Set mysteries in specific historical periods with accurate details of the time. Include period-appropriate investigation methods and social constraints. Research historical context thoroughly to create authentic atmosphere.",
    "LOCKED_ROOM": "Present seemingly impossible crimes where the perpetrator could not have committed the crime and escaped. Focus on logical solutions to apparently supernatural or impossible situations. Challenge readers to solve the puzzle.",
    "POLICE_PROCEDURAL": "Detail realistic police investigation methods and procedures. Include team dynamics, forensic science, and bureaucratic challenges. Focus on how law enforcement actually works while solving crimes.",
    "PSYCHOLOGICAL_THRILLER": "Focus on the mental and emotional aspects of crime and investigation. Explore the psychology of both criminals and investigators. Include unreliable narrators, mind games, and character studies.",
  },
  "NEW_ADULT": {
    "CAREER_BEGINNINGS": "Center on characters starting their professional lives and navigating early career challenges. Create realistic portrayal of entry-level positions, workplace dynamics, and professional identity formation. Develop characters balancing career ambitions with other life aspects. Consider both setbacks and successes in early professional development. Balance workplace specifics with broader life transitions.",
    "COLLEGE_FICTION": "Set primarily in college or university environments with characters 18-22. Create authentic portrayal of contemporary college life and academic pressures. Develop characters balancing studies, social life, and self-discovery. Consider diversity of college experiences beyond stereotypical party scenes. Balance campus-specific elements with universal coming-of-age themes.",
    "COMING_OF_INDEPENDENCE": "Focus specifically on the transition from family dependence to self-sufficiency. Create realistic portrayal of early independence including financial challenges, living arrangements, and decision-making. Develop characters learning from mistakes and growing in capability. Consider both freedoms and responsibilities of independence. Balance struggles with moments of achievement and growth.",
    "CONTEMPORARY_NEW_ADULT": "Focus on experiences of 18-25 year olds transitioning to independent adulthood. Create protagonists dealing with college, first jobs, or leaving home. Develop realistic portrayal of early adult challenges and decisions. Consider themes of identity formation and taking responsibility. Balance coming-of-age elements with more mature content than YA.",
    "INTERNATIONAL_NEW_ADULT": "Focus on early adult experiences (18-25) in specific cultural contexts outside North America. Research cultural specifics of early adulthood in the chosen setting. Develop authentic portrayal of how adulthood transition varies across cultures. Consider both universal aspects of growing up and culturally specific expectations. Balance cultural specificity with accessible narrative for diverse readers.",
    "NEW_ADULT_FANTASY": "Create fantasy worlds with protagonists in early adulthood (18-25) facing age-appropriate challenges. Develop magical elements that metaphorically reflect early adult transitions and choices. Consider how fantasy elements interact with themes of independence and identity formation. Balance worldbuilding with character development relevant to target age group.",
    "NEW_ADULT_LITERARY": "Create literary-quality writing focused on the early adult experience (18-25). Develop complex characters with rich interior lives navigating the transition to adulthood. Focus on nuanced exploration of identity formation and life choices. Consider both universal and generation-specific aspects of early adulthood. Balance literary techniques with accessible narrative.",
    "NEW_ADULT_ROMANCE": "Center on romantic relationships during the transitional years of early adulthood (18-25). Create age-appropriate romantic situations and challenges. Develop characters navigating both romantic relationships and other early adult responsibilities. Consider realistic portrayal of physical relationships for this age group. Balance romance with identity development and life transitions.",
    "NEW_ADULT_SCIENCE_FICTION": "Develop science fiction scenarios with protagonists in early adulthood (18-25). Create technological or scientific elements that connect to early adult experiences and choices. Consider how science fiction concepts can illuminate questions of identity and purpose. Balance technical elements with character-driven narrative appropriate for target age.",
    "NEW_ADULT_THRILLER": "Create suspenseful situations with protagonists in early adulthood (18-25). Develop threats and challenges appropriate to early adult life circumstances. Consider how limited life experience affects characters' responses to danger. Balance intensity and mature content with age-appropriate scenarios. Focus on both external threats and psychological tension.",
  },
  "ROMANCE": {
    "CONTEMPORARY_ROMANCE": "Set in present day with realistic characters facing modern relationship challenges. Focus on emotional development and romantic tension. Include obstacles that are relevant to contemporary life. Ensure satisfying romantic resolution.",
    "EROTIC_ROMANCE": "Include explicit romantic scenes as integral parts of character and relationship development. Focus on emotional connection alongside physical attraction. Ensure romantic plot drives the narrative.",
    "HISTORICAL_ROMANCE": "Set in specific historical periods with accurate cultural and social details. Research historical constraints on relationships and social expectations. Balance historical authenticity with romantic fantasy.",
    "INSPIRATIONAL_ROMANCE": "Include faith-based themes and clean romantic content. Focus on spiritual growth alongside romantic development. Avoid explicit content while maintaining emotional depth.",
    "LGBTQ_ROMANCE": "Center romantic relationships within LGBTQ+ communities. Include authentic representation and avoid stereotypes. Focus on universal themes of love while acknowledging specific challenges faced by LGBTQ+ individuals.",
    "PARANORMAL_ROMANCE": "Include supernatural elements such as vampires, werewolves, angels, or psychic abilities. Balance romantic development with fantastical elements. Explore how supernatural aspects affect romantic relationships.",
    "ROMANTIC_SUSPENSE": "Combine romance with thriller or mystery elements. Include danger that threatens the romantic protagonists. Balance romantic development with suspenseful plot advancement. Ensure both romance and suspense are equally important.",
    "SPORTS_ROMANCE": "Center around athletic characters and sports environments. Research specific sports for authentic details. Include themes of teamwork, dedication, and overcoming physical and emotional challenges.",
  },
  "SCIENCE_FICTION": {
    "ALIEN_CONTACT": "Center on first contact or ongoing relationships with extraterrestrial beings. Explore communication barriers, cultural differences, and the impact on human society. Consider both peaceful and hostile scenarios. Examine how contact changes humanity's perspective.",
    "CYBERPUNK": "Set in a high-tech, low-life dystopian future. Feature advanced technology like cybernetic enhancements, artificial intelligence, and virtual reality alongside social disorder, corporate dominance, and urban decay. Protagonists are often marginalized, anti-heroic characters navigating corrupt systems.",
    "DYSTOPIAN": "Create oppressive societies where individual freedom is restricted. Focus on protagonists who challenge or become aware of systemic control. Explore themes of surveillance, conformity, and resistance. Balance the bleakness with hope and human resilience.",
    "HARD_SCIENCE_FICTION": "Emphasize scientific accuracy and plausibility. Ground fantastical elements in real scientific principles. Include detailed explanations of technology and its implications. Focus on how scientific advancement affects characters and society.",
    "POST_APOCALYPTIC": "Set in a world after a major catastrophe. Focus on survival, rebuilding society, and human adaptation to harsh new realities. Explore themes of hope, community, and what it means to be human in extreme circumstances. Balance despair with resilience.",
    "SPACE_OPERA": "Focus on grand-scale adventures across galaxies. Include interstellar travel, multiple alien civilizations, epic conflicts, and advanced technology. Emphasize character-driven narratives against the backdrop of a vast universe. Balance action with philosophical themes about humanity's place in the cosmos.",
    "STEAMPUNK": "Set in an alternative Victorian era with steam-powered technology. Combine historical elements with fantastical mechanical inventions. Feature brass, gears, goggles, and elaborate contraptions. Balance adventure with social commentary on industrialization.",
    "TIME_TRAVEL": "Explore the consequences and paradoxes of temporal manipulation. Consider how changes to the past affect the present and future. Develop clear rules for time travel in your world. Focus on character development through experiencing different time periods.",
  },
  "THRILLER": {
    "ACTION_THRILLER": "Emphasize fast-paced action sequences and physical confrontations. Include chase scenes, fights, and high-stakes situations. Balance action with character development and plot advancement.",
    "DOMESTIC_THRILLER": "Focus on ordinary people in extraordinary threatening situations, often involving family or close relationships. Include psychological manipulation and betrayal by trusted individuals.",
    "ESPIONAGE": "Feature spies, intelligence agencies, and international intrigue. Include realistic espionage techniques and geopolitical tensions. Balance action with psychological complexity of living double lives.",
    "LEGAL_THRILLER": "Set in legal environments with lawyers, judges, or law students as protagonists. Include courtroom drama, legal procedures, and justice system complexities. Research legal processes for authenticity.",
    "MEDICAL_THRILLER": "Feature medical professionals or settings with health-related threats. Include accurate medical terminology and procedures. Explore ethical dilemmas in healthcare and medical conspiracies.",
    "POLITICAL_THRILLER": "Involve government conspiracies, political intrigue, or international relations. Include realistic political processes and motivations. Explore themes of power, corruption, and institutional conflict.",
    "PSYCHOLOGICAL_THRILLER": "Focus on mental and emotional suspense rather than physical action. Include unreliable narrators, mind games, and exploration of characters' psychological states. Build tension through uncertainty about reality and motivations.",
    "TECHNO_THRILLER": "Include cutting-edge technology as central to the plot. Feature cyber warfare, artificial intelligence, or advanced scientific concepts. Research current technological trends for plausibility.",
  },
  "WESTERN": {
    "CATTLE_DRIVE": "Focus on the challenges and adventures of moving cattle across long distances. Include themes of perseverance, teamwork, and man versus nature. Research historical cattle drive practices.",
    "CONTEMPORARY_WESTERN": "Set in modern American West with traditional Western themes adapted to contemporary settings. Include ranch life, small town dynamics, and conflicts between tradition and progress.",
    "MARSHAL_SHERIFF": "Center on law enforcement officers bringing order to frontier communities. Include themes of duty, justice, and the challenges of maintaining law in lawless territories.",
    "MINING_TOWN": "Set in boom towns built around mining operations. Include themes of sudden wealth, lawlessness, and community building in harsh conditions. Explore the boom and bust cycle of frontier economics.",
    "OUTLAW": "Focus on criminals and those living outside the law in Western settings. Explore themes of rebellion against authority and survival outside civilization. Include moral ambiguity and complex motivations.",
    "REVISIONIST_WESTERN": "Challenge traditional Western tropes and present more realistic or critical views of the frontier period. Include perspectives of Native Americans, women, and minorities often overlooked in traditional Westerns.",
    "TRADITIONAL_WESTERN": "Set in American frontier period (1860s-1890s) with cowboys, outlaws, and frontier towns. Include themes of justice, survival, and civilization versus wilderness. Focus on moral codes and individual honor.",
    "WEIRD_WESTERN": "Combine Western settings with supernatural, science fiction, or horror elements. Include supernatural creatures, time travel, or advanced technology in frontier settings.",
  },
  "WOMENS_FICTION": {
    "CONTEMPORARY_WOMENS_FICTION": "Focus on modern women's experiences, relationships, and personal growth. Create complex female protagonists facing contemporary challenges. Develop nuanced relationships between women (friendships, family, colleagues). Consider issues particularly relevant to women's lives without reducing characters to issues. Balance challenges with moments of empowerment and connection.",
    "WOMENS_LITERARY_FICTION": "Create literary-quality writing focused on women's experiences and perspectives. Develop complex female characters with rich interior lives. Focus on nuanced exploration of themes particularly relevant to women. Consider both universal human experiences and gender-specific perspectives. Balance literary techniques with accessible, engaging narrative.",
  },
  "YOUNG_ADULT": {
    "BOARDING_SCHOOL": "Set in residential educational institutions with focus on student life, friendships, and institutional dynamics. Include themes of independence, peer relationships, and academic pressure.",
    "CONTEMPORARY_YA": "Focus on teenage protagonists facing realistic modern challenges such as school, family, relationships, and identity formation. Include authentic teen voice and concerns. Address issues relevant to young adults.",
    "YA_DYSTOPIAN": "Set in oppressive future societies with teenage protagonists challenging or surviving systemic control. Include themes of rebellion, identity, and finding one's place in an unjust world.",
    "YA_FANTASY": "Include magical elements with teenage protagonists discovering their powers or roles in fantastical worlds. Balance fantasy adventure with coming-of-age themes. Include themes of destiny and self-discovery.",
    "YA_HISTORICAL": "Set in historical periods with teenage protagonists experiencing major historical events. Research how historical events specifically affected young people. Include coming-of-age themes within historical context.",
    "YA_ROMANCE": "Center romantic relationships between teenage characters. Include emotional intensity and idealization common in first love. Balance romance with other aspects of teenage life and development.",
    "YA_SCIENCE_FICTION": "Include scientific or technological elements with teenage protagonists. Focus on how technology affects young people's lives and relationships. Include themes of progress versus tradition.",
    "YA_THRILLER": "Create suspenseful situations with teenage protagonists facing danger. Include age-appropriate threats and challenges. Focus on survival and problem-solving skills of young adults.",
  },
}
`;
